### DISCUSSION TITLE
Help for beginners to find and hide an element, CSS classes

### DISCUSSION THREAD

**Discussion by Chunkst** [seleniumbase/SeleniumBase#2468] [Q&A] [answered]
Hi, I'm new to SeleniumBase and I'm trying to automate a specific action. I don't understand much despite the documentation... I've managed to do what I wanted in terms of user actions, but now I'd like to hide a popup when a site opens. However, in spite of numerous attempts, it's impossible to retrieve the element with the CSS classes to hide it... Thanks in advance for your help

```
h=Driver()
h.get('https://a.random.url/')
u=h.find_element('button')
# for _ in username: _.style='display: none'
print(u)
```

**Comment by mdmintz** [comment, marked as answer]
Hello! Here's a sample script that hides various elements on a web page using the `SB()` format that has more methods than the `Driver()` format: (`sleep()` added so that you have time to see the difference).

```python
from seleniumbase import SB

with SB() as sb:
    sb.open("https://fr.wikipedia.org/wiki/Tour_Eiffel")
    sb.sleep(1)
    sb.hide_elements("div#siteNotice")
    sb.sleep(1)
    sb.hide_elements("nav#mw-panel-toc")
    sb.sleep(1)
    sb.hide_elements('img[src*="Tour_Eiffel_Wikimedia"]')
    sb.sleep(2)
```

To hide only the first matching selector found, use `hide_element(selector)` instead of `hide_elements(selector)`.

For the full French experience, here's a SeleniumBase example using the French Language API: (Adds `--demo` to highlight elements)

```python
from seleniumbase.translate.french import CasDeBase
CasDeBase.main(__name__, __file__, "--demo")

class MaClasseDeTest(CasDeBase):
    def test_exemple_1(self):
        self.ouvrir("https://fr.wikipedia.org/wiki/")
        self.vérifier_texte("Wikipédia")
        self.vérifier_élément('[alt="Wikipédia"]')
        self.js_taper("#searchform input", "Crème brûlée")
        self.cliquer("#searchform button")
        self.vérifier_texte("Crème brûlée", "#firstHeading")
        self.vérifier_élément('img[alt*="Crème brûlée"]')
        self.js_taper("#searchform input", "Jardin des Tuileries")
        self.cliquer("#searchform button")
        self.vérifier_texte("Jardin des Tuileries", "#firstHeading")
        self.vérifier_élément('img[alt*="Jardin des Tuileries"]')
        self.retour()
        self.vérifier_url_contient("brûlée")
        self.en_avant()
        self.vérifier_url_contient("Jardin")
```

Translations can be found in [SeleniumBase/examples/translations](https://github.com/seleniumbase/SeleniumBase/tree/master/examples/translations).

All methods can be found here: [SeleniumBase/help_docs/method_summary.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md).

(The `Driver()`-specific methods are a smaller subset of that, at the bottom of the `method_summary.md` file.)

See [SeleniumBase/examples](https://github.com/seleniumbase/SeleniumBase/tree/master/examples) for all examples.

**Reply by Chunkst** [reply]
Thank you very much! However I still do not understand some points (: ... On the one hand now the web instance closes itself (whereas I want it to stay open until I tell it to close manually). On the other hand, my code doesn't work without the 'with', which I don't understand either... Thanks in advance ;)
Here is a part of it :
```
h=SB();x,y=lambda:h.close(),lambda m:h.sleep(m)
h.open(f'https://my.url/{c[p]}/')
h.hide_elements('button[class$="py-3 px-6 w-full"]')
def e()->None:
	if is_pressed('q'): h.quit();startfile(R);system('taskkill /f /im py.exe')
	h.refresh();h.execute_cdp_cmd('Storage.clearDataForOrigin',{"origin":'*',
		"storageTypes":'all'});h.click('button[data-v-8775d804]')
	h.switch_to.window(h.window_handles[-1]);x()
	h.switch_to.window(h.window_handles[0])
	y(3);h.type('#firstName',f'{choice(f)}')
	h.click('button[class$="py-3 px-6 w-full"]');y(8);l=h.get_text('body')
```

**Reply by mdmintz** [reply]
The `with` indicates a Python context manager. Actions are performed before the indented block starts, and after the indented block ends.

To pause the script in the middle (which will keep the browser open) use `breakpoint()`.

**Reply by Chunkst** [reply]
Okok super I did not know. Thank you so much for your speed! I still have a few questions though haha. Is Seleniumbase a more advanced version of Selenium? 
Given the automation script I sent in my previous message, is it enough for me to put everything in a with? Can I create a function in a with? I don't really understand the concept of context manager, sorry about that...

**Reply by Chunkst** [reply]
in fact, my function is more complete than the one above and is recursive, repeating until it detects a keypress

**Reply by mdmintz** [reply]
SeleniumBase is a more advanced version of Selenium. You can call functions from the “with” block. With the “breakpoint()”, you can continue your script by typing “c” and pressing ENTER from the console.

**Reply by Chunkst** [reply]
Super bravo and thanks again! One last thing, what is the equivalent of switch_to.window for SB()?

**Reply by Chunkst** [reply]
Oh, and I still can't hide a single element despite numerous attempts... I've tested CSS selectors that worked for other methods like clicking... I don't understand why nothing works.

```
with SB() as h:
	x,y=lambda:h.close(),lambda m:h.sleep(m)
	h.open(f'https://a.random.url/{c[p]}/')
	h.hide_elements('input#email')
```

**Reply by mdmintz** [reply]
`switch_to_window(1)` will switch the window in SeleniumBase.

If your “input” element is in an iframe, you’ll need to switch into the iframe first. You can also call `sb.driver` to access driver methods.
